### DISCUSSION TITLE
Feature Request: center to a coordinate via command line arguments

### DISCUSSION THREAD

**Discussion by markusd112** [Stellarium/stellarium#2135] [Ideas]
Hi,

I want to start Stellarium from another application and center the view to a given RA / DEC coordinate. I haven't find a way to give these parameters via command line arguments. Wouldn't it be a good idea to enable the command line interface to accept these parameters? 

Thanks

markus

**Comment by gzotti** [comment]
We could add the CLI switches if this is all you need and if you can accept the startup time every time you call it. However, interacting from another program is best done with the RemoteControl interface.

**Reply by markusd112** [reply]
Thanks. I am looking for a simple way to open Stellarium and change the view to a specific RA/DEC-Coordinate. A simple way would be via commandline arguments, as it is implemented e.g. in Carte du Ciel:

https://www.ap-i.net/skychart/en/documentation/line_commands

Is there a way to use the RemoteControl Plugin to let stellarium move the view to a specific RA/DEC-Coordinate? I haven't found anything in the manual... Calling an URL with RA/DEC as URL-parameter would be an easy way too for my use case.

I have a simple database for astrophotograpy and I want to give the user the option to show the object from the database in stellarium. So I need some way to control stellarium from an external application (here it is e.g. MS Access).

Thanks.

Markus

**Reply by alex-w** [reply]
Please check this [URL](https://stellarium.org/doc/0.21/remoteControlDoc.html)

**Reply by gzotti** [reply]
https://stellarium.org/doc/head/group__remoteControl.html

**Reply by markusd112** [reply]
Thanks. I tried to read/understand the documentation with the links you sent, but I haven't managed it.... I don't understand, which URL I have to call and what is the right format for RA/DEC to pass in the URL to control Sterllarium to move to the desired coordinate? I 

I tried with

http://localhost:8090/api/search

...but this URL seems not to be the right one.

I am not very familiar with all the internal programming structures of stellarium. For a novice like me it is very difficult to get it work based on the documentation... Thanks a lot.

**Reply by gzotti** [reply]
Try first on the commandline, e.g.: 
`curl --data 'j2000=[0.5,0.3,0.2]' http://localhost:8090/api/main/view
`
The x/y/z values were just anything. The conversion of spherical to Cartesian x/y/z coordinates and variants of this call (e.g. jNow) is described in https://stellarium.org/doc/head/remoteControlApi.html

I cannot say how you establish an HTTP connection in your programming environment

**Reply by markusd112** [reply]
Establishing an HTTP-Connection is not the problem. In the moment I try it with my browser. It seems that the URL 

http://localhost:8090/api/main/view/

gives back the actual view focus of stellarium. It seems not to be intended to SET the position of the view... when calling http://localhost:8090/api/main/view?j2000=[0.5,0.3,0.2]   only the actual position is displayed by the browser in JSON format, but the view isn't set...
p.s.: only to be sure we talk about the same: I want to center the view to a given RA/DEC coordinate (not to set location of the observer)

Thanks a lot for your help!!

**Reply by gzotti** [reply]
Yes, there is a GET and a POST operation type. curl --data... does POST, curl -G --data ... does GET. You must send a POST request.

**Reply by markusd112** [reply]
Sorry, my fault. Thanks! I will try it via curl on my linux machine
